xdp/src/device.rs

```diff
@@ -180,25 +180,39 @@ impl NetworkDevice {
             return Err(io::Error::last_os_error());
         }
 
-        Ok((rp.rx_pending as usize, rp.tx_pending as usize))
+        Ok(RingSizes {
+            rx: rp.rx_pending as usize,
+            tx: rp.tx_pending as usize,
+        })
+    }
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct RingSizes {
+    pub rx: usize,
+    pub tx: usize,
+}
+
+impl Default for RingSizes {
+    fn default() -> Self {
+        // These are reasonable defaults for devices which don't have a set ring size
```

Comment:
might be good to indicate these need to be power of 2

Reply:
done